### ISSUE TITLE
About SFT performance

### ISSUE THREAD

**Issue by m-Just** [Deep-Agent/R1-V#72]
Hi, thank you for open-sourcing such great work! I have some questions about the SFT performance shown in the readme:

1. Is the SFT model trained with [Clevr_CoGenT_TrainA_R1](https://huggingface.co/datasets/MMInstruction/Clevr_CoGenT_TrainA_R1), which includes (1) problems harder than "How many items are there in the image?" and (2) the reasoning trace of DeepSeek-R1?
2. If so, wouldn't it be too hard for the 2B model to learn these, resulting in the poor SFT performance?
3. Have you tried SFT the model with just the answers for the "How many items are there in the image?" problems, without the reasoning traces?

I want to ask these because the significant gap between SFT and GRPO performance is a bit puzzling to me. In fact, from the 2nd question of #5, we can see that the reasoning traces of the GRPO model are trivial. Based on this, it seems to me that SFT with just the answers (w/o the reasoning traces) may give us a similar level of performance as GRPO on this dataset?

**Comment by chenllliang** [comment]
In the SFT experiment shown in the chart, we employed chain-of-thought supervised fine-tuning (CoT-SFT) using CoT-form data as described in the readme. Our later subsequent experiments revealed that simple counting tasks do not benefit from the chain-of-thought approach just as you said. In fact, vanilla SFT outperformed CoT-SFT in out-of-distribution (OOD) testing, though both methods still achieved lower performance than GRPO when trained for the same number of steps.

**Comment by austingg** [comment]
@chenllliang Maybe you can try `InternVL-MPO` series which claim for better CoT MLLM ability.

**Comment by m-Just** [comment]
@chenllliang Thanks for your reply! Actually, I just found that the pre-trained model (Qwen2-VL-2B-Instruct) can achieve ~86.5% accuracy on SuperCLEVR with this simple prompt:
```python
QUESTION_TEMPLATE = "{Question} Answer with a single number."
```
and with the answer extraction pattern changed accordingly to
```python
answer_pattern = r'(\d+)'
```
So, the pre-trained model is already pretty good at this counting task, and GRPO with reasoning traces is not really helping (yet?). I would recommend adding this detail to the readme because the real improvement is not as dramatic as it seems.

In this context, I guess more complex problems such as GEOQA are indeed more interesting to study. Good move!

**Comment by EchoDreamer** [comment]
> [@chenllliang](https://github.com/chenllliang) Thanks for your reply! Actually, I just found that the pre-trained model (Qwen2-VL-2B-Instruct) can achieve **~86.5%** accuracy on SuperCLEVR with this simple prompt:
> 
> QUESTION_TEMPLATE = "{Question} Answer with a single number."
> and with the answer extraction pattern changed accordingly to
> 
> answer_pattern = r'(\d+)'
> So, the pre-trained model is already pretty good at this counting task. Meanwhile, GRPO with reasoning traces is not really helping (yet?). I would recommend adding this detail to the readme because the real improvement is not as dramatic as it seems.
> 
> In this context, I guess more complex problems such as GEOQA are indeed more interesting to study. Good move!

**Comment by EchoDreamer** [comment]
> [@chenllliang](https://github.com/chenllliang) Thanks for your reply! Actually, I just found that the pre-trained model (Qwen2-VL-2B-Instruct) can achieve **~86.5%** accuracy on SuperCLEVR with this simple prompt:
> 
> QUESTION_TEMPLATE = "{Question} Answer with a single number."
> and with the answer extraction pattern changed accordingly to
> 
> answer_pattern = r'(\d+)'
> So, the pre-trained model is already pretty good at this counting task. Meanwhile, GRPO with reasoning traces is not really helping (yet?). I would recommend adding this detail to the readme because the real improvement is not as dramatic as it seems.
> 
> In this context, I guess more complex problems such as GEOQA are indeed more interesting to study. Good move!

Indeed, in provided setting, I got 87% accuracy! Need to try more difficult experiments.

**Comment by chenllliang** [comment]
@austingg @m-Just @EchoDreamer Thanks for your attention. We add this detail to the readme that letting the base model output cot would harm the counting ability. We also find that in the GEOQA task, letting the base model output cot first would have 8%  increase, which is an interesting difference to the counting experiments. We also summaize more experiments and findings in our latest [post](https://deepagent.notion.site/rlvr-in-vlms).
